controller/controller.go

```diff
@@ -87,10 +89,17 @@ func (c Controller) Reconcile(ctx context.Context, list []types.Resource) (resul
 			if patch == nil {
 				patch = client.MergeFrom(c.object.DeepCopyObject().(client.Object))
 			}
+			kinds, _, err := (*c.scheme).ObjectKinds(res.Runtime())
+			if err != nil || len(kinds) == 0 {
+				log.Error().Caller().Err(err).Msg("unable to get object kind")
+				return ctrl.Result{}, err
+			}
```

Comment:
The only problem we will face is if the same Kinds have more than one API version,

Do you know if we pass in a kind as "text" to scheme, will it return what API version? If so; we should use this so we can avoid saving APIVersion.

Reply:
Issue is resolved in the latest commit. We get all known types in the scheme and get only the types with the same kind and group as the missing resource. We then get a list of every version of the resource and delete them all